src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php

```diff
@@ -11,134 +11,15 @@
 
 namespace Symfony\Component\HttpKernel\EventListener;
 
-use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
-use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
-use Symfony\Component\HttpKernel\Event\PostResponseEvent;
-use Symfony\Component\HttpKernel\KernelEvents;
-use Symfony\Component\HttpKernel\Profiler\Profiler;
-use Symfony\Component\HttpFoundation\RequestMatcherInterface;
-use Symfony\Component\HttpFoundation\RequestStack;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Profiler\EventListener\HttpProfilerListener;
 
 /**
  * ProfilerListener collects data for the current request by listening to the kernel events.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ * @deprecated since x.x, to be removed in x.x. Use Symfony\Component\Profiler\EventListener\ProfileListener instead.
  */
-class ProfilerListener implements EventSubscriberInterface
+class ProfilerListener extends HttpProfilerListener
```

Comment:
Because I would like to make it possible to create a ConsoleProfileListener for console commands.

if you noticed I also renamed Profiler to HttpProfiler for this reason, and I think it's best to keep these names in sync to avoid confusion.